### PR TITLE
107 enhance api doc

### LINKED ISSUE #107: Add informationen for response code to REST-API documentation
Response codes for errors are not specified or described for the REST-API endpoints.

Example annotation for `/api/v1/import`

```
@ApiResponses(value = { 
  @ApiResponse(responseCode = "201", description = "Import successful. Document created", 
    content = { @Content(mediaType = "application/json", 
      schema = @Schema(implementation = EntityCreateResponse.class)) }),
  @ApiResponse(responseCode = "422", description = "Import failed. Document with the same id exists or invalid document", 
    content = @Content), 
  @ApiResponse(responseCode = "503", description = "Validation failed", 
    content = @Content) })
```

### PR DESCRIPTION
**Pull request by mfd2007**
Add missing informations about response code to the OpenAPI documentation.

This PR fixes #107 .

### CODE REVIEW

File: src/main/java/de/bsi/secvisogram/csaf_cms_backend/rest/AdvisoryController.java [hunk 1]
```diff
@@ -390,6 +657,34 @@ public ResponseEntity<JsonNode> readTemplate(
...
+            description = "Bad format requested." 
+          ),
+        @ApiResponse(
+          responseCode = "404", 
+          description = "Advisory not found." 
+        ),
+        @ApiResponse(
+          responseCode = "401", 
+          description = "Unauthorized access."
+        ),
```
Comment (tschmidtb51) [inline src/main/java/de/bsi/secvisogram/csaf_cms_backend/rest/AdvisoryController.java:682]:
```suggestion
        @ApiResponse(
          responseCode = "401", 
          description = "Unauthorized access."
        ),
        @ApiResponse(
          responseCode = "404", 
          description = "Advisory not found." 
        ),

File: src/main/java/de/bsi/secvisogram/csaf_cms_backend/rest/AdvisoryController.java [hunk 2]
```diff
@@ -454,6 +749,27 @@ static MediaType determineExportResponseContentType(@Nullable final ExportFormat
...
+          ),
+        @ApiResponse(
+            responseCode = "400", 
+            description = "Advisory ID not found." 
+          ),
+        @ApiResponse(
+          responseCode = "401", 
+          description = "Unauthorized access to change workflow state."
+        )
+      })
```
Comment (tschmidtb51) [inline src/main/java/de/bsi/secvisogram/csaf_cms_backend/rest/AdvisoryController.java:772]: Is it only `formatted` or is the advisory invalid itself?
```suggestion
        @ApiResponse(
            responseCode = "400", 
            description = "Advisory ID not found." 
          ),
        @ApiResponse(
            responseCode = "401", 
            description = "Unauthorized access to change workflow state."
        ),
        @ApiResponse(
            responseCode = "422", 
            description = "Invalid formatted advisory." 
        )
      })

File: src/main/java/de/bsi/secvisogram/csaf_cms_backend/rest/AdvisoryController.java [hunk 3]
```diff
@@ -498,6 +814,27 @@ private ResponseEntity<String> changeWorkflowState(String advisoryId, String rev
...
+            description = "Invalid formated advisory." 
+          ),
+        @ApiResponse(
+            responseCode = "400", 
+            description = "Advisory ID not found." 
+          ),
+        @ApiResponse(
+          responseCode = "401", 
+          description = "Unauthorized access to change workflow state."
+        )
```
Comment (tschmidtb51) [inline src/main/java/de/bsi/secvisogram/csaf_cms_backend/rest/AdvisoryController.java:836]: same here

File: src/main/java/de/bsi/secvisogram/csaf_cms_backend/rest/AdvisoryController.java [hunk 4]
```diff
@@ -525,6 +862,27 @@ public ResponseEntity<String> setWorkflowStateToReview(
...
```
Comment (tschmidtb51) [inline src/main/java/de/bsi/secvisogram/csaf_cms_backend/rest/AdvisoryController.java:884]: and here

File: src/main/java/de/bsi/secvisogram/csaf_cms_backend/rest/AdvisoryController.java [hunk 5]
```diff
@@ -553,6 +911,31 @@ public ResponseEntity<String> setWorkflowStateToApproved(
...
+            description = "Invalid formated advisory." 
+          ),
+        @ApiResponse(
+            responseCode = "400", 
+            description = "Advisory ID not found." 
+          ),
+        @ApiResponse(
+          responseCode = "401", 
+          description = "Unauthorized access to change workflow state."
+        ),
```
Comment (tschmidtb51) [inline src/main/java/de/bsi/secvisogram/csaf_cms_backend/rest/AdvisoryController.java:933]: and here

File: src/main/java/de/bsi/secvisogram/csaf_cms_backend/rest/AdvisoryController.java [hunk 6]
```diff
@@ -577,8 +960,34 @@ public ResponseEntity<String> setWorkflowStateToRfPublication(
...
```
Comment (tschmidtb51) [inline src/main/java/de/bsi/secvisogram/csaf_cms_backend/rest/AdvisoryController.java:985]: see above

File: src/main/java/de/bsi/secvisogram/csaf_cms_backend/rest/AdvisoryController.java [hunk 7]
```diff
@@ -200,9 +346,46 @@ public ResponseEntity<EntityCreateResponse> importCsafDocument(@RequestBody Json
...
+              mediaType = MediaType.APPLICATION_JSON_VALUE,
+              schema = @Schema(
+                  implementation = EntityUpdateResponse.class
+              )
+            )
+          }
+        ),
+        @ApiResponse(
+            responseCode = "400", 
+            description = "Database error."
```
Comment (tschmidtb51) [inline src/main/java/de/bsi/secvisogram/csaf_cms_backend/rest/AdvisoryController.java:370]: Database error? vs. Bad Request?

**Review by tschmidtb51 — CHANGES_REQUESTED** [review]
It looks like the HTTP status codes have different messages - was that intended?

File: src/main/java/de/bsi/secvisogram/csaf_cms_backend/rest/AdvisoryController.java (hunk 3 above)
Comment (mfd2007) [reply src/main/java/de/bsi/secvisogram/csaf_cms_backend/rest/AdvisoryController.java]: Changed order.

File: src/main/java/de/bsi/secvisogram/csaf_cms_backend/rest/AdvisoryController.java (hunk 5 above)
Comment (mfd2007) [reply src/main/java/de/bsi/secvisogram/csaf_cms_backend/rest/AdvisoryController.java]: Changed order.

File: src/main/java/de/bsi/secvisogram/csaf_cms_backend/rest/AdvisoryController.java (hunk 7 above)
Comment (mfd2007) [reply src/main/java/de/bsi/secvisogram/csaf_cms_backend/rest/AdvisoryController.java]: Clarified description. It's not simply a database error. It is a malformed submitted document.

**Comment by mfd2007** [comment]
Additional information according unauthorized access on workflow change was intended.

File: src/main/java/de/bsi/secvisogram/csaf_cms_backend/rest/AdvisoryController.java [hunk 8]
```diff
@@ -525,6 +862,27 @@ public ResponseEntity<String> setWorkflowStateToReview(
...
+          responseCode = "400", 
+          description = "Advisory ID not found." 
+        ),
+        @ApiResponse(
+          responseCode = "401", 
+          description = "Unauthorized access to change workflow state."
+        ),
+        @ApiResponse(
+          responseCode = "422", 
+          description = "Invalid formated advisory."
```
Comment (tschmidtb51) [inline src/main/java/de/bsi/secvisogram/csaf_cms_backend/rest/AdvisoryController.java:883]:
```suggestion
          description = "Invalid formatted advisory."

File: src/main/java/de/bsi/secvisogram/csaf_cms_backend/rest/AdvisoryController.java [hunk 9]
```diff
@@ -553,6 +911,31 @@ public ResponseEntity<String> setWorkflowStateToApproved(
...
```
Comment (tschmidtb51) [inline src/main/java/de/bsi/secvisogram/csaf_cms_backend/rest/AdvisoryController.java:932]:
```suggestion
          description = "Invalid formatted advisory."

File: src/main/java/de/bsi/secvisogram/csaf_cms_backend/rest/AdvisoryController.java [hunk 10]
```diff
@@ -577,8 +960,34 @@ public ResponseEntity<String> setWorkflowStateToRfPublication(
...
```
Comment (tschmidtb51) [inline src/main/java/de/bsi/secvisogram/csaf_cms_backend/rest/AdvisoryController.java:984]:
```suggestion
          description = "Invalid formatted advisory."

**Review by tschmidtb51 — CHANGES_REQUESTED** [review]
Some `t`s missing ;-)

**Comment by tschmidtb51** [comment]
BTW: I couldn't figure out why the tests are failing...

**Comment by mfd2007** [comment]
Problems with test are documented in #124 and do not relate to changes in this PR.

**Review by tschmidtb51 — APPROVED** [review]
LGTM

**Pull request merged**